org.jacoco.core.test/src/org/jacoco/core/instr/InstrumenterTest.java

```diff
@@ -71,51 +66,20 @@ public String toString() {
 
 	}
 
-	private static final class AccessorGenerator
-			implements IExecutionDataAccessorGenerator {
+	private SystemPropertiesRuntime runtime;
 
-		long classId;
-
-		public int generateDataAccessor(final long classId,
-				final String classname, final int probeCount,
-				final MethodVisitor mv) {
-			this.classId = classId;
-			InstrSupport.push(mv, probeCount);
-			mv.visitIntInsn(Opcodes.NEWARRAY, Opcodes.T_BOOLEAN);
-			return 1;
-		}
-
-	}
-
-	private AccessorGenerator accessorGenerator;
 	private Instrumenter instrumenter;
 
 	@Before
 	public void setup() throws Exception {
-		accessorGenerator = new AccessorGenerator();
-		instrumenter = new Instrumenter(accessorGenerator);
-	}
-
-	@Test
-	public void should_instrument_java10_class() throws Exception {
-		final byte[] originalBytes = createClass(BytecodeVersion.V10);
-		final byte[] bytes = new byte[originalBytes.length];
-		System.arraycopy(originalBytes, 0, bytes, 0, originalBytes.length);
-		final long expectedClassId = CRC64.classId(bytes);
-
-		final byte[] instrumentedBytes = instrumenter.instrument(bytes, "");
-
-		assertArrayEquals(originalBytes, bytes);
-		assertEquals(BytecodeVersion.V10,
-				BytecodeVersion.get(instrumentedBytes));
-		assertEquals(expectedClassId, accessorGenerator.classId);
```

Comment:
This removes test that was preventing disappearance of

https://github.com/jacoco/jacoco/blob/68ab19f3463bf5076485185fe2abe5ec8a726679/org.jacoco.core/src/org/jacoco/core/instr/Instrumenter.java#L105

i.e. test that was preventing mistake that we did with Java 9 - https://github.com/jacoco/jacoco/pull/636

Reply:
@marchof and this reminded me about https://gitlab.ow2.org/asm/asm/merge_requests/143#note_34290 :

> ```java
> public class MyClassReader extends ClassReader {
> 
>   public MyClassReader(byte[] classFile, int version) {
>     super(setVersion(classFile, version));
>     // Restore the original version.
>     for (int i = 0; i < 4; ++i) classFile[i + 4] = classFile[i];
>     // Optional: restore the 0xCAFEBABE magic header.
>   }
> 
>   private static byte[] setVersion(byte[] classFile, int version) {
>     // Copy the original version over the 0xCAFEBABE magic header.
>     for (int i = 0; i < 4; ++i) classFile[i] = classFile[i + 4];
>     // Change the version to the given one.
>     classFile[4] = (byte) (version >>> 24);
>     classFile[5] = (byte) (version >>> 16);
>     classFile[6] = (byte) (version >>> 8);
>     classFile[7] = (byte) version;
>     return classFile;
>   }
> }
> ```

what might be safer/easier to use than our `BytecodeVersion.downgradeIfNeeded` followed by `BytecodeVersion.set`.